hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ASSL.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.s3a;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.contract.ContractTestUtils;
+import org.apache.hadoop.security.ssl.OpenSSLSocketFactory;
+
+import org.apache.hadoop.util.NativeCodeLoader;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.apache.hadoop.fs.contract.ContractTestUtils.dataset;
+import static org.apache.hadoop.fs.contract.ContractTestUtils.writeDataset;
+import static org.junit.Assume.assumeTrue;
+
+/**
+ * Tests non-default values for {@link Constants#SSL_CHANNEL_MODE}.
+ */
+public class ITestS3ASSL extends AbstractS3ATestBase {
+
+  @Test
+  public void testOpenSSL() throws IOException {
+    assumeTrue(NativeCodeLoader.buildSupportsOpenssl());
+    Configuration conf = new Configuration();
+    conf.setEnum(Constants.SSL_CHANNEL_MODE,
+            OpenSSLSocketFactory.SSLChannelMode.OpenSSL);
+    S3AFileSystem fs = S3ATestUtils.createTestFileSystem(conf);
```

Comment:
use in try-with-resources to avoid leaking FS instances

Reply:
done